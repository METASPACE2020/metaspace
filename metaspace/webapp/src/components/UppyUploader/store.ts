// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
