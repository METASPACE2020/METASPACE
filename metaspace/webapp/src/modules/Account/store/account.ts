import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
