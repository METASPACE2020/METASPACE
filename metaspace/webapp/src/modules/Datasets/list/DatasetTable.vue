\<template>
  <div>
    <filter-panel level="dataset" />

    <div>
      <el-form
        id="dataset-list-header"
        :inline="true"
      >
        <el-radio-group
          value="List"
          size="small"
          @input="onChangeTab"
        >
          <el-radio-button label="List" />
          <el-radio-button label="Summary" />
        </el-radio-group>

        <el-checkbox-group
          v-model="categories"
          :min="1"
          class="dataset-status-checkboxes"
        >
          <el-checkbox
            class="cb-started"
            label="started"
          >
            Processing {{ count('started') }}
          </el-checkbox>
          <el-checkbox
            class="cb-queued"
            label="queued"
          >
            Queued {{ count('queued') }}
          </el-checkbox>
          <el-checkbox label="finished">
            Finished {{ count('finished') }}
          </el-checkbox>
          <el-checkbox
            v-if="canSeeFailed"
            class="cb-failed"
            label="failed"
          >
            Failed {{ count('failed') }}
          </el-checkbox>
        </el-checkbox-group>
        <div style="flex-grow: 1;" />

        <el-button
          v-if="nonEmpty"
          :disabled="isExporting"
          size="small"
          @click="startExport"
        >
          Export to CSV
        </el-button>
      </el-form>
    </div>

    <dataset-list
      :datasets="datasets"
      allow-double-column
    />
  </div>
</template>

<script>
import {
  datasetDetailItemsQuery,
  datasetCountQuery,
  datasetDeletedQuery,
  datasetStatusUpdatedQuery,
} from '../../../api/dataset'
import { metadataExportQuery } from '../../../api/metadata'
import DatasetList from './DatasetList.vue'
import { FilterPanel } from '../../Filters/index'
import FileSaver from 'file-saver'
import delay from '../../../lib/delay'
import formatCsvRow, { csvExportHeader, formatCsvTextArray } from '../../../lib/formatCsvRow'
import { currentUserRoleQuery } from '../../../api/user'
import { removeDatasetFromAllDatasetsQuery } from '../../../lib/updateApolloCache'
import { sortBy, uniqBy } from 'lodash-es'
import updateApolloCache from '../../../lib/updateApolloCache'

const processingStages = ['started', 'queued', 'failed', 'finished']

export default {
  name: 'DatasetTable',
  components: {
    DatasetList,
    FilterPanel,
  },
  data() {
    return {
      recordsPerPage: 10,
      csvChunkSize: 1000,
      categories: ['started', 'queued', 'finished'],
      isExporting: false,
    }
  },

  computed: {
    noFilters() {
      const df = this.$store.getters.filter
      for (var key in df)
        if (df[key]) return false
      return true
    },

    nonEmpty() {
      return this.datasets.length > 0
    },

    allDatasets() {
      let list = []
      for (const category of processingStages) {
        if (this[category]) {
          list = list.concat(this[category])
        }
      }
      // Sort again in case any datasets have had their status changed and are now in the wrong list
      const sortOrder = ['FAILED', 'ANNOTATING', 'QUEUED', 'FINISHED']
      list = uniqBy(list, 'id')
      list = sortBy(list, dataset => sortOrder.indexOf(dataset.status))
      return list
    },

    datasets() {
      return this.allDatasets
        .filter(ds =>
          (ds.status === 'FAILED' && this.categories.includes('failed'))
           || (ds.status === 'ANNOTATING' && this.categories.includes('started'))
           || (ds.status === 'QUEUED' && this.categories.includes('queued'))
           || (ds.status === 'FINISHED' && this.categories.includes('finished')))
    },

    canSeeFailed() {
      return this.currentUser != null && this.currentUser.role === 'admin'
    },
  },

  apollo: {
    $subscribe: {
      datasetDeleted: {
        query: datasetDeletedQuery,
        result({ data }) {
          const datasetId = data.datasetDeleted.id;
          ['failed', 'finished', 'queued', 'started'].forEach(queryName => {
            removeDatasetFromAllDatasetsQuery(this, queryName, datasetId)
          })
        },
      },
      datasetStatusUpdated: {
        query: datasetStatusUpdatedQuery,
        async result({ data }) {
          const { dataset, action, stage, is_new: isNew } = data.datasetStatusUpdated
          if (dataset != null && action === 'ANNOTATE' && stage === 'QUEUED' && isNew) {
            updateApolloCache(this, 'queued', oldVal => {
              return {
                ...oldVal,
                allDatasets: oldVal.allDatasets && [dataset, ...oldVal.allDatasets],
              }
            })
          }
        },
      },
    },

    currentUser: {
      query: currentUserRoleQuery,
      fetchPolicy: 'cache-first',
    },

    failed: {
      fetchPolicy: 'cache-and-network',
      query: datasetDetailItemsQuery,
      update: data => data.allDatasets,
      skip() {
        return !this.canSeeFailed
      },
      variables() {
        return this.queryVariables('FAILED')
      },
    },

    started: {
      fetchPolicy: 'cache-and-network',
      query: datasetDetailItemsQuery,
      update: data => data.allDatasets,
      variables() {
        return this.queryVariables('ANNOTATING')
      },
    },

    queued: {
      fetchPolicy: 'cache-and-network',
      query: datasetDetailItemsQuery,
      update: data => data.allDatasets,
      variables() {
        return this.queryVariables('QUEUED')
      },
    },

    finished: {
      fetchPolicy: 'cache-and-network',
      query: datasetDetailItemsQuery,
      update: data => data.allDatasets,
      variables() {
        return this.queryVariables('FINISHED')
      },
    },

    finishedCount: {
      fetchPolicy: 'cache-and-network',
      query: datasetCountQuery,
      update: data => data.countDatasets,
      variables() {
        return this.queryVariables('FINISHED')
      },
    },
  },

  methods: {
    formatSubmitter: (row, col) =>
      row.submitter.name,
    formatDatasetName: (row, col) =>
      row.name.split('//', 2)[1],
    formatResolvingPower: (row, col) =>
      (row.analyzer.resolvingPower / 1000).toFixed(0) * 1000,

    queryVariables(status) {
      const body = {
        dFilter: Object.assign({ status }, this.$store.getters.gqlDatasetFilter),
        query: this.$store.getters.ftsQuery,
        inpFdrLvls: [],
        checkLvl: 10,
      }
      if (status === 'FINISHED') { body.inpFdrLvls = [10] }
      return body
    },

    count(stage) {
      let count = null
      // assume not too many items are failed/queued/annotating so they are all visible in the web app,
      // but check all lists because they may be in the wrong list due to status updates after they were loaded
      if (stage === 'failed')
        count = this.allDatasets.filter(ds => ds.status === 'FAILED').length
      if (stage === 'queued')
        count = this.allDatasets.filter(ds => ds.status === 'QUEUED').length
      if (stage === 'started')
        count = this.allDatasets.filter(ds => ds.status === 'ANNOTATING').length
      if (stage === 'finished') {
        const inOtherLists = this.allDatasets.filter(ds => ds.status === 'FINISHED').length
           - (this.finished && this.finished.length || 0)
        count = this.finishedCount == null ? null : this.finishedCount + inOtherLists
      }

      return count != null && !isNaN(count) ? `(${count})` : ''
    },

    async startExport() {
      let csv = csvExportHeader()

      csv += formatCsvRow(['datasetId', 'datasetName', 'group', 'submitter',
        'PI', 'organism', 'organismPart', 'condition', 'growthConditions', 'ionisationSource',
        'maldiMatrix', 'analyzer', 'resPower400', 'polarity', 'uploadDateTime',
        'FDR@10%', 'database', 'opticalImage',
      ])

      function person(p) { return p ? p.name : '' }

      function formatRow(row) {
        const { groupApproved, group, principalInvestigator } = row
        return formatCsvRow([
          row.id,
          row.name,
          groupApproved && group ? group.shortName : '',
          person(row.submitter),
          principalInvestigator ? person(principalInvestigator)
            : groupApproved && group ? (group.adminNames || []).join(', ')
              : '',
          row.organism,
          row.organismPart,
          row.condition,
          row.growthConditions,
          row.ionisationSource,
          row.maldiMatrix,
          row.analyzer.type,
          Math.round(row.analyzer.resolvingPower),
          row.polarity.toLowerCase(),
          row.uploadDateTime,
          row.fdrCounts ? row.fdrCounts.counts : '',
          row.fdrCounts ? row.fdrCounts.dbName : '',
          (row.rawOpticalImageUrl) ? window.location.origin + row.rawOpticalImageUrl : 'No optical image',
        ])
      }

      function writeCsvChunk(rows) {
        for (const row of rows) {
          csv += formatRow(row)
        }
      }

      this.isExporting = true
      const self = this

      const v = this.queryVariables('FINISHED')
      const chunks = []
      let offset = 0

      v.limit = this.csvChunkSize

      while (self.isExporting && offset < self.finishedCount) {
        const variables = Object.assign(v, { offset })
        const resp = await self.$apollo.query({ query: metadataExportQuery, variables })

        offset += this.csvChunkSize
        writeCsvChunk(resp.data.datasets)
        await delay(50)
      }

      if (!self.isExporting)
        return

      self.isExporting = false

      const blob = new Blob([csv], { type: 'text/csv; charset="utf-8"' })
      FileSaver.saveAs(blob, 'metaspace_datasets.csv')
    },

    onChangeTab(tab) {
      this.$store.commit('setDatasetTab', tab)
    },
  },
}
</script>

<style lang="scss">

 #dataset-page {
   display: flex;
   justify-content: center;
 }

 .dataset-status-checkboxes {
   padding: 5px 20px;
   display: flex;
   flex-direction: row;
   align-items: center;
 }

 .cb-started .el-checkbox__input.is-checked .el-checkbox__inner {
   background: #5eed5e;
 }

 .cb-queued .el-checkbox__input.is-checked .el-checkbox__inner {
   background: #72c8e5;
 }

 .cb-failed .el-checkbox__input.is-checked .el-checkbox__inner {
   background: #f56c6c;
 }

 #dataset-list-header {
   display: flex;
   flex-wrap: wrap;
   margin-bottom: 10px;
 }
</style>
