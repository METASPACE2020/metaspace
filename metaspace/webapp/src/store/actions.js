import { fetchOptionListsQuery } from '../api/metadata'
import { decodeParams } from '../modules/Filters'
import { computed } from 'vue'
import store from '../store/index'

export default {
  async initFilterLists(context, apolloClient) {
    if (context.state.filterListsLoading || context.state.filterLists != null) return
    await context.commit('setFilterListsLoading')

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    })

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {
      ...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds),
    })

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)

    context.commit('updateFilter', filter)
  },
}
