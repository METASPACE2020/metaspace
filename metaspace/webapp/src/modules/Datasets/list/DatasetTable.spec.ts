import { nextTick, ref } from 'vue'
import { afterEach, vi } from 'vitest'
import DatasetTable from './DatasetTable.vue'
import * as FileSaver from 'file-saver'
import { merge } from 'lodash-es'
import router from '../../../router'
import store from '../../../store'
import { DefaultApolloClient, useQuery } from '@vue/apollo-composable'
import { flushPromises, mount } from '@vue/test-utils'
import { initMockGraphqlClient } from '../../../tests/utils/mockGraphqlClient'

vi.mock('file-saver', () => ({
  saveAs: vi.fn(),
}))

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(),
  useMutation: vi.fn(),
  useSubscription: vi.fn(() => ({ onResult: vi.fn() })),
  DefaultApolloClient: vi.fn(),
}))

let graphqlMocks: any

const blobToText = (blob: Blob) =>
  new Promise<string>((resolve, reject) => {
    const reader = new FileReader()
    reader.onload = () => resolve(reader.result as string)
    reader.onerror = (e) => reject(e.target.error)
    reader.readAsText(blob)
  })

describe('DatasetTable', () => {
  const mockMetadataJson = JSON.stringify({
    Metadata_Type: 'Imaging MS',
    MS_Analysis: {
      Detector_Resolving_Power: { mz: 1234, Resolving_Power: 123456 },
    },
  })
  const mockFdrCounts = {
    databaseId: 6,
    dbName: 'HMDB',
    dbVersion: 'v2.5',
    levels: [10],
    counts: [20],
  }
  const mockDataset = {
    id: 'REPLACEME',
    status: 'FINISHED',
    metadataJson: mockMetadataJson,
    description: null,
    databases: [
      { name: 'CHEBI', version: '', id: 2 },
      { name: 'HMDB', version: 'v2.5', id: 6 },
      { name: 'HMDB', version: 'v4', id: 22 },
    ],
    polarity: 'POSITIVE',
    fdrCounts: mockFdrCounts,
    groupApproved: true,
    canEdit: false,
    canDelete: false,
  }
  const mockGraphql = async (queryParams) => {
    graphqlMocks = await initMockGraphqlClient({
      Query: () => queryParams,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(queryParams).reduce((acc, key) => ({ ...acc, [key]: queryParams[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
      subscribeToMore: vi.fn(),
      refetch: vi.fn(),
    })
  }

  afterEach(() => {
    vi.useRealTimers()
  })

  it('should match snapshot', async () => {
    const queryParams = {
      allDatasets: () => {
        return [
          { ...mockDataset, id: 'ANNOTATING1', status: 'ANNOTATING' },
          { ...mockDataset, id: 'QUEUED1', status: 'QUEUED' },
          { ...mockDataset, id: 'FINISHED1', status: 'FINISHED' },
        ]
      },
      countDatasetsPerGroup: () => ({
        counts: [
          { fieldValues: ['QUEUED'], count: 2 },
          { fieldValues: ['ANNOTATING'], count: 1 },
          { fieldValues: ['FINISHED'], count: 20 },
        ],
      }),
    }
    await mockGraphql(queryParams)

    const wrapper = mount(DatasetTable, {
      global: {
        plugins: [router, store],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })

    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be able to export a CSV', async () => {
    const queryParams = {
      allDatasets: (_: any, params: any) => {
        const offset = params?.offset || 0
        const filter = params?.filter
        const status = filter?.status || 'FINISHED'

        return [
          merge({}, mockDataset, { principalInvestigator: null, id: `${status}1`, status }),
          merge({}, mockDataset, { principalInvestigator: null, groupApproved: false, id: `${status}2`, status }),
          merge({}, mockDataset, {
            principalInvestigator: null,
            group: { adminNames: ['group', 'admin', 'names'] },
            id: `${status}3`,
            status,
          }),
          merge({}, mockDataset, {
            principalInvestigator: { name: 'principal investigator' },
            id: `${status}4`,
            status,
          }),
        ].slice(offset, offset + 2)
      },
      countDatasets: () => 4,
    }
    await mockGraphql(queryParams)

    const wrapper = mount(DatasetTable, {
      global: {
        plugins: [router, store],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })

    wrapper.vm.state.csvChunkSize = 2
    await new Promise((resolve) => setTimeout(resolve, 1))

    await flushPromises()
    await nextTick()

    await (wrapper.vm as any).startExport()
    expect(FileSaver.saveAs).toBeCalled()
    const blob: Blob = (FileSaver.saveAs as any).mock.calls[0][0]
    const csv = await blobToText(blob)
    const csvWithoutDateHeader = csv.replace(/# Generated at .*\n/, '')

    expect(csvWithoutDateHeader).toMatchSnapshot()
  })
})
