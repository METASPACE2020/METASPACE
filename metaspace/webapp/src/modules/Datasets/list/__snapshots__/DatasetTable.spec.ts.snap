// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetTable should be able to export a CSV 1`] = `
"# URL: http://localhost/
\\"datasetId\\",\\"datasetName\\",\\"group\\",\\"submitter\\",\\"PI\\",\\"organism\\",\\"organismPart\\",\\"condition\\",\\"growthConditions\\",\\"ionisationSource\\",\\"maldiMatrix\\",\\"analyzer\\",\\"resPower400\\",\\"polarity\\",\\"uploadDateTime\\",\\"FDR@10%\\",\\"database\\",\\"opticalImage\\"
\\"FINISHED1\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"datasets.0.group.adminNames, datasets.0.group.adminNames\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"20\\",\\"HMDB-v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED2\\",\\"datasets.1.name\\",\\"\\",\\"datasets.1.submitter.name\\",\\"\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"20\\",\\"HMDB-v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
\\"FINISHED3\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"group, admin, names\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"20\\",\\"HMDB-v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED4\\",\\"datasets.1.name\\",\\"datasets.1.group.shortName\\",\\"datasets.1.submitter.name\\",\\"principal investigator\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"20\\",\\"HMDB-v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
"
`;

exports[`DatasetTable should match snapshot 1`] = `
<div>
  <div class="filter-panel">
    <mock-el-select placeholder="Add filter" class="filter-select">
      <mock-el-option value="group" label="Select group"></mock-el-option>
      <mock-el-option value="project" label="Select project"></mock-el-option>
      <mock-el-option value="submitter" label="Select submitter"></mock-el-option>
      <mock-el-option value="datasetIds" label="Select dataset"></mock-el-option>
      <mock-el-option value="compoundName" label="Search molecule"></mock-el-option>
      <mock-el-option value="polarity" label="Select polarity"></mock-el-option>
      <mock-el-option value="organism" label="Select organism"></mock-el-option>
      <mock-el-option value="organismPart" label="Select organism part"></mock-el-option>
      <mock-el-option value="condition" label="Select organism condition"></mock-el-option>
      <mock-el-option value="growthConditions" label="Select sample growth conditions"></mock-el-option>
      <mock-el-option value="analyzerType" label="Select analyzer"></mock-el-option>
      <mock-el-option value="ionisationSource" label="Select ionisation source"></mock-el-option>
      <mock-el-option value="maldiMatrix" label="Select MALDI matrix"></mock-el-option>
      <mock-el-option value="simpleQuery" label="Search anything"></mock-el-option>
      <mock-el-option value="metadataType" label="Select data type"></mock-el-option>
    </mock-el-select>
  </div>
  <div>
    <form class="el-form el-form--inline" id="dataset-list-header">
      <div role="radiogroup" class="el-radio-group"><label role="radio" aria-checked="true" tabindex="0" class="el-radio-button el-radio-button--small is-active"><input type="radio" tabindex="-1" class="el-radio-button__orig-radio" value="List"><span class="el-radio-button__inner">List</span></label> <label role="radio" tabindex="-1" class="el-radio-button el-radio-button--small"><input type="radio" tabindex="-1" class="el-radio-button__orig-radio" value="Summary"><span class="el-radio-button__inner">Summary</span></label></div>
      <div role="group" aria-label="checkbox-group" class="el-checkbox-group dataset-status-checkboxes" mock-v-loading="0"><label class="el-checkbox cb-annotating is-checked"><span class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value="ANNOTATING"></span><span class="el-checkbox__label">
          Processing (1)
        <!----></span></label> <label class="el-checkbox cb-queued is-checked"><span class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value="QUEUED"></span><span class="el-checkbox__label">
          Queued (2)
        <!----></span></label> <label class="el-checkbox is-checked"><span class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span><input type="checkbox" aria-hidden="false" class="el-checkbox__original" value="FINISHED"></span><span class="el-checkbox__label">
          Finished (20)
        <!----></span></label>
        <!---->
      </div>
      <div style="flex-grow: 1;"></div> <button type="button" class="el-button el-button--default el-button--small">
        <!---->
        <!----><span>
        Export to CSV
      </span></button>
    </form>
  </div>
  <div class="dataset-list allow-double-column" mock-v-loading="0">
    <div class="dataset-item border border-solid border-gray-200 leading-5">
      <mock-el-dialog title="Provided metadata">
        <div class="el-row">
          <mock-el-tree id="metadata-tree" data="[object Object],[object Object]" node-key="id" default-expanded-keys="MS analysis,Detector resolving power,Data Management"></mock-el-tree>
        </div>
      </mock-el-dialog>
      <div class="opt-image--container">
        <div class="thumb"><img src="allDatasets.1.thumbnailOpticalImageUrl" alt="Optical image" class="thumb--img" style="transform: scaleY(1) translateY(0%); height: 100px;"> <img src="allDatasets.1.ionThumbnailUrl" alt="Ion image thumbnail" class="thumb--img thumb--img__ion" style="transform: scaleY(1) translateY(0%); height: 100px;">
          <!---->
        </div>
      </div>
      <div class="ds-info">
        <div class="ds-item-line flex"><span title="allDatasets.1.name" class="font-bold truncate">allDatasets.1.name</span>
          <mock-el-popover trigger="hover" placement="top" class="ml-1">
            <div slot-key="default">
              <div mock-v-loading="true">

              </div>
            </div>
            <div slot-key="reference"><i class="el-icon-lock"></i></div>
          </mock-el-popover>
        </div>
        <div class="ds-item-line text-gray-700"><span title="Filter by species" class="ds-add-filter">
        allDatasets.1.organism</span>,
          <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.1.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.1.condition)</span></div>
        <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.1.ionisationSource</span> +
          <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.1.analyzer.type</span>,
          <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>,
          RP 123k @ 1234
        </div>
        <div class="ds-item-line">
          Submitted <mock-elapsed-time date=""></mock-elapsed-time> by
          <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.1.submitter.name</span><span>,
        <mock-el-dropdown show-timeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="text-base text-primary cursor-pointer">
            allDatasets.1.group.shortName
          </span> </div>
        <div slot-key="dropdown">
          <mock-el-dropdown-menu>
            <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
            <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
          </mock-el-dropdown-menu>
        </div>
        </mock-el-dropdown></span>
      </div>
      <!---->
    </div>
    <div class="ds-actions">
      <!---->
      <!----> <span><div title="Waiting in the queue" class="striped-progressbar queued"></div></span>
      <div><i class="el-icon-view"></i> <a href="#">Show full metadata</a></div>
      <!---->
      <div class="ds-download"><i class="el-icon-download"></i> <a href="#">Download</a></div>
      <!---->
      <!---->
    </div>
    <!---->
  </div>
  <div class="dataset-item border border-solid border-gray-200 leading-5">
    <mock-el-dialog title="Provided metadata">
      <div class="el-row">
        <mock-el-tree id="metadata-tree" data="[object Object],[object Object]" node-key="id" default-expanded-keys="MS analysis,Detector resolving power,Data Management"></mock-el-tree>
      </div>
    </mock-el-dialog>
    <div class="opt-image--container">
      <div class="thumb"><img src="allDatasets.0.thumbnailOpticalImageUrl" alt="Optical image" class="thumb--img" style="transform: scaleY(1) translateY(0%); height: 100px;"> <img src="allDatasets.0.ionThumbnailUrl" alt="Ion image thumbnail" class="thumb--img thumb--img__ion" style="transform: scaleY(1) translateY(0%); height: 100px;">
        <!---->
      </div>
    </div>
    <div class="ds-info">
      <div class="ds-item-line flex"><span title="allDatasets.0.name" class="font-bold truncate">allDatasets.0.name</span>
        <!---->
      </div>
      <div class="ds-item-line text-gray-700"><span title="Filter by species" class="ds-add-filter">
        allDatasets.0.organism</span>,
        <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.0.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.0.condition)</span></div>
      <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.0.ionisationSource</span> +
        <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.0.analyzer.type</span>,
        <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>,
        RP 123k @ 1234
      </div>
      <div class="ds-item-line">
        Submitted <mock-elapsed-time date=""></mock-elapsed-time> by
        <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.0.submitter.name</span><span>,
        <mock-el-dropdown show-timeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="text-base text-primary cursor-pointer">
            allDatasets.0.group.shortName
          </span> </div>
      <div slot-key="dropdown">
        <mock-el-dropdown-menu>
          <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
          <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
        </mock-el-dropdown-menu>
      </div>
      </mock-el-dropdown></span>
    </div>
    <!---->
  </div>
  <div class="ds-actions">
    <!----> <span><div title="Processing is under way" class="striped-progressbar processing"></div></span>
    <!---->
    <div><i class="el-icon-view"></i> <a href="#">Show full metadata</a></div>
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
</div>
<div class="dataset-item border border-solid border-gray-200 leading-5">
  <mock-el-dialog title="Provided metadata">
    <div class="el-row">
      <mock-el-tree id="metadata-tree" data="[object Object],[object Object]" node-key="id" default-expanded-keys="MS analysis,Detector resolving power,Data Management"></mock-el-tree>
    </div>
  </mock-el-dialog>
  <div class="opt-image--container">
    <div class="thumb"><img src="allDatasets.2.thumbnailOpticalImageUrl" alt="Optical image" class="thumb--img" style="transform: scaleY(1) translateY(0%); height: 100px;"> <img src="allDatasets.2.ionThumbnailUrl" alt="Ion image thumbnail" class="thumb--img thumb--img__ion" style="transform: scaleY(1) translateY(0%); height: 100px;">
      <!---->
    </div>
  </div>
  <div class="ds-info">
    <div class="ds-item-line flex"><span title="allDatasets.2.name" class="font-bold truncate">allDatasets.2.name</span>
      <!---->
    </div>
    <div class="ds-item-line text-gray-700"><span title="Filter by species" class="ds-add-filter">
        allDatasets.2.organism</span>,
      <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.2.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.2.condition)</span></div>
    <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.2.ionisationSource</span> +
      <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.2.analyzer.type</span>,
      <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>,
      RP 123k @ 1234
    </div>
    <div class="ds-item-line">
      Submitted <mock-elapsed-time date=""></mock-elapsed-time> by
      <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.2.submitter.name</span><span>,
        <mock-el-dropdown show-timeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="text-base text-primary cursor-pointer">
            allDatasets.2.group.shortName
          </span> </div>
    <div slot-key="dropdown">
      <mock-el-dropdown-menu>
        <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
        <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
      </mock-el-dropdown-menu>
    </div>
    </mock-el-dropdown></span>
  </div>
  <div class="ds-item-line"><span><a href="/annotations?db=HMDB-v2.5&amp;ds=FINISHED1&amp;mdtype=allDatasets.2.metadataType" class="">20 annotations</a>
        @ FDR 10% (HMDB-v2.5)
      </span></div>
</div>
<div class="ds-actions"><span><i class="el-icon-picture"></i> <mock-el-popover trigger="hover" placement="top"><div slot-key="default"><div class="db-link-list">
          Select a database:
          <div><a href="/annotations?db=HMDB-v2.5&amp;ds=FINISHED1&amp;mdtype=allDatasets.2.metadataType" class="">
              HMDB-v2.5
            </a></div><div><a href="/annotations?db=HMDB-v4&amp;ds=FINISHED1&amp;mdtype=allDatasets.2.metadataType" class="">
              HMDB-v4
            </a></div><div><a href="/annotations?db=CHEBI&amp;ds=FINISHED1&amp;mdtype=allDatasets.2.metadataType" class="">
              CHEBI
            </a></div></div> </div><div slot-key="reference"><a>Browse annotations</a></div></mock-el-popover> <br></span>
  <!---->
  <!---->
  <div><i class="el-icon-view"></i> <a href="#">Show full metadata</a></div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>
<!---->
</div>
<!---->
</div>
</div>
`;
